Launcher requires undeclared tsconfig-paths dependency

**Describe the bug**

The gauge-ts 0.5.0 launcher unconditionally preloads `tsconfig-paths/register`, but `tsconfig-paths` is neither declared as a dependency of the `gauge-ts` npm package nor documented as a required project dependency.

As a result, projects without a direct `tsconfig-paths` dependency fail to start:

```text
Error: Cannot find module 'tsconfig-paths/register'
Require stack:
- internal/preload
```

**To Reproduce**

1. Download `gauge-ts-0.5.0.zip` from the gauge-ts GitHub Release and install it:

   ```shell
   gauge install ts --file gauge-ts-0.5.0.zip
   ```

2. Create a TypeScript Gauge project using gauge-ts 0.5.0 without adding `tsconfig-paths` to `package.json`.

3. Install the project dependencies:

   ```shell
   npm install
   ```

4. Run the specifications:

   ```shell
   gauge run specs
   ```

**Expected behavior**

Projects that do not use TypeScript path aliases should run without requiring `tsconfig-paths`.

If `tsconfig-paths` is intended to be mandatory, it should be declared and documented so that newly initialized projects work without additional manual setup.

**Screenshots**

Not applicable.

**Desktop (please complete the following information):**

- OS: Windows
- Gauge and plugin version:

  ```text
  Gauge version: 1.6.33
  Commit Hash: c3fbfeb

  Plugins
  -------
  ts (0.5.0)
  ```

- Node version: `v22.22.3`
- npm version: `10.9.8`

**Additional context**

The original support added in #44 loaded `tsconfig-paths/register` only when the module was installed.

The conditional loading remained in place after #181. During #208, commit `268aa02e1be85b5780379dd4110b4db58bcfa5ee` changed the launcher to preload it unconditionally as part of an e2e path-alias fix:

```js
"ts-node",
"--esm",
"-r",
"tsconfig-paths/register",
```

The `hasModule()` helper still exists in `launcher.mjs` but is now unused, which may indicate that unconditional loading was not intended for every project.

The gauge-ts e2e project declares `tsconfig-paths` directly, so the e2e tests do not expose this failure.

Possible solutions include restoring conditional loading or declaring `tsconfig-paths` as a runtime dependency and loading it in a way that also works with strict dependency layouts such as pnpm and Yarn Plug'n'Play.

If the maintainers agree that conditional loading is the intended behavior, I'd be happy to submit a PR with the fix and regression coverage.

--- a/gauge-ts/launcher-runner.cjs
+++ b/gauge-ts/launcher-runner.cjs
@@ -20,4 +20,16 @@ function getPackageRunner(value = process.env.GAUGE_TS_PACKAGE_RUNNER) {
   return runner;
 }
 
-module.exports = { getPackageRunner };
+function getTsNodeArgs({ hasTsconfigPaths, useShell }) {
+  const shellQuote = useShell ? '"' : "";
+
+  return [
+    "ts-node",
+    "--esm",
+    ...(hasTsconfigPaths ? ["-r", "tsconfig-paths/register"] : []),
+    "-e",
+    `${shellQuote}import { start } from 'gauge-ts/dist/RunnerServer'; start();${shellQuote}`,
+  ];
+}
+
+module.exports = { getPackageRunner, getTsNodeArgs };

--- a/gauge-ts/launcher.mjs
+++ b/gauge-ts/launcher.mjs
@@ -13,7 +13,7 @@ if (Number.parseInt(version[0]) < 20) {
 import { spawn } from "node:child_process";
 import launcherRunner from "./launcher-runner.cjs";
 
-const { getPackageRunner } = launcherRunner;
+const { getPackageRunner, getTsNodeArgs } = launcherRunner;
 
 const { GAUGE_PROJECT_ROOT } = process.env;
 
@@ -28,16 +28,10 @@ function hasModule(name) {
 
 function startCommand() {
   const useShell = process.platform === "win32";
-  const shellQuote = useShell ? '"' : "";
-
-  const opts = [
-    "ts-node",
-    "--esm",
-    "-r",
-    "tsconfig-paths/register",
-    "-e",
-    `${shellQuote}import { start } from 'gauge-ts/dist/RunnerServer'; start();${shellQuote}`,
-  ];
+  const opts = getTsNodeArgs({
+    hasTsconfigPaths: hasModule("tsconfig-paths"),
+    useShell,
+  });
 
   const [command, ...runnerArgs] = getPackageRunner();
   const runner = spawn(command, [...runnerArgs, ...opts], {

--- a/gauge-ts/package.json
+++ b/gauge-ts/package.json
@@ -1,6 +1,6 @@
 {
   "name": "gauge-ts",
-  "version": "0.5.0",
+  "version": "0.5.1",
   "description": "Typescript runner for Gauge",
   "main": "./dist/index.js",
   "types": "./dist/index.d.ts",

--- a/gauge-ts/tests/LauncherTests.ts
+++ b/gauge-ts/tests/LauncherTests.ts
@@ -1,4 +1,4 @@
-const { getPackageRunner } = require("../launcher-runner.cjs");
+const { getPackageRunner, getTsNodeArgs } = require("../launcher-runner.cjs");
 
 describe("launcher package runner", () => {
   test.each([
@@ -20,4 +20,36 @@ describe("launcher package runner", () => {
       );
     },
   );
+});
+
+describe("ts-node arguments", () => {
+  test("preloads tsconfig-paths when it is installed", () => {
+    expect(getTsNodeArgs({ hasTsconfigPaths: true, useShell: false })).toEqual([
+      "ts-node",
+      "--esm",
+      "-r",
+      "tsconfig-paths/register",
+      "-e",
+      "import { start } from 'gauge-ts/dist/RunnerServer'; start();",
+    ]);
+  });
+
+  test("does not preload tsconfig-paths when it is not installed", () => {
+    expect(getTsNodeArgs({ hasTsconfigPaths: false, useShell: false })).toEqual(
+      [
+        "ts-node",
+        "--esm",
+        "-e",
+        "import { start } from 'gauge-ts/dist/RunnerServer'; start();",
+      ],
+    );
+  });
+
+  test("quotes the eval script when using a shell", () => {
+    const args = getTsNodeArgs({ hasTsconfigPaths: false, useShell: true });
+
+    expect(args.at(-1)).toBe(
+      "\"import { start } from 'gauge-ts/dist/RunnerServer'; start();\"",
+    );
+  });
 });

--- a/gauge-ts/ts.json
+++ b/gauge-ts/ts.json
@@ -16,6 +16,6 @@
     "linux": ["./launcher.mjs", "--start"],
     "windows": ["launcher.bat", "--start"]
   },
-  "version": "0.5.0",
+  "version": "0.5.1",
   "gRPCSupport": true
 }

